src/function/scalar/list/list_lambdas.cpp

```diff
@@ -0,0 +1,418 @@
+			break;
+		}
+	}
+
+	// compute the new lengths and offsets, and create a selection vector
+	for (idx_t i = 0; i < elem_cnt; i++) {
+
+		while (lists_len[appended_lists_cnt] == 0) {
+			result_entries[appended_lists_cnt].offset = curr_list_offset;
+			result_entries[appended_lists_cnt].length = 0;
+			appended_lists_cnt++;
+			if (appended_lists_cnt == lists_len.size()) {
+				break;
+			}
+		}
+
+		// found a true value
+		if (lambda_validity.RowIsValid(i)) {
+			if (lambda_values[i] > 0) {
+				true_sel.set_index(true_count++, i);
+				curr_list_len++;
+			}
+		}
+		curr_original_list_len++;
+
+		if (lists_len[appended_lists_cnt] == curr_original_list_len) {
+			result_entries[appended_lists_cnt].offset = curr_list_offset;
+			result_entries[appended_lists_cnt].length = curr_list_len;
+			curr_list_offset += curr_list_len;
+			appended_lists_cnt++;
+			curr_list_len = 0;
+			curr_original_list_len = 0;
+		}
+	}
+
+	// slice to get the new lists and append them to the result
+	Vector new_lists(input_chunk.data[0], true_sel, true_count);
+	new_lists.Normalify(true_count);
+	VectorData new_lists_child_data;
+	new_lists.Orrify(true_count, new_lists_child_data);
+	ListVector::Append(result, new_lists, *new_lists_child_data.sel, true_count, 0);
+}
+
+static void ExecuteExpression(vector<LogicalType> &types, vector<LogicalType> &result_types, idx_t &elem_cnt,
+                              SelectionVector &sel, vector<SelectionVector> &sel_vectors, DataChunk &input_chunk,
+                              DataChunk &lambda_chunk, Vector &child_vector, DataChunk &args,
+                              ExpressionExecutor &expr_executor) {
+
+	input_chunk.InitializeEmpty(types);
+
+	// set the list child vector
+	Vector slice(child_vector, sel, elem_cnt);
+	slice.Normalify(elem_cnt);
+	input_chunk.data[0].Reference(slice);
+
+	// set the other vectors
+	vector<Vector> slices;
+	for (idx_t col_idx = 0; col_idx < args.ColumnCount() - 1; col_idx++) {
+		slices.emplace_back(Vector(args.data[col_idx + 1], sel_vectors[col_idx], elem_cnt));
+		slices[col_idx].Normalify(elem_cnt);
+		input_chunk.data[col_idx + 1].Reference(slices[col_idx]);
+	}
+	input_chunk.SetCardinality(elem_cnt);
+
+	// initialize the lambda result chunk
+	lambda_chunk.Initialize(result_types);
+	lambda_chunk.SetCardinality(elem_cnt);
+
+	// execute the lambda expression
+	expr_executor.Execute(input_chunk, lambda_chunk);
+}
+
+template <bool IS_TRANSFORM = true>
+static void ListLambdaFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+
+	// always at least the list argument
+	D_ASSERT(args.ColumnCount() >= 1);
+
+	auto count = args.size();
+	Vector &lists = args.data[0];
+
+	result.SetVectorType(VectorType::FLAT_VECTOR);
+	auto result_entries = FlatVector::GetData<list_entry_t>(result);
+	auto &result_validity = FlatVector::Validity(result);
+
+	if (lists.GetType().id() == LogicalTypeId::SQLNULL) {
+		result_validity.SetInvalid(0);
```

Comment:
Do we need to set the result to a constant here?

Reply:
I am actually not sure... I didn't do that for the other list functions, either. But could already be wrong there?